services/webhook/deliver.go

```diff
@@ -170,18 +171,32 @@ func Deliver(ctx context.Context, t *webhook_model.HookTask) error {
 		}
 	}()
 
+	// OK We're now ready to attempt to deliver the task - we must double check that it
+	// has not been delivered in the meantime
+	marked, err := webhook_model.MarkTaskDelivered(ctx, t)
```

Comment:
But if you have marked but then you sent failed?

Reply:
Yes. is_delivered  is always set to true from this point onwards see the defer